caikit_nlp/modules/text_classification/sequence_classification.py

```diff
@@ -0,0 +1,235 @@
+        device: Union[str, int, None],
+    ):
+        super().__init__()
+        error.type_check(
+            "<NLP74125820E>",
+            PreTrainedTokenizerBase,
+            tokenizer=tokenizer,
+        )
+        error.type_check(
+            "<NLP64751996E>",
+            PreTrainedModel,
+            model=model,
+        )
+        self.tokenizer = tokenizer
+        self.model = model
+        self.device = device
+
+    ################################## API functions #############################################
+
+    def run(self, text: str) -> ClassificationResult:
+        """Run the sequence classification, truncates sequences too long for model
+
+        Args:
+            text: str
+                Input string to be classified
+
+        Returns:
+            ClassificationResult
+        """
+        scores_dict = self._get_scores(text)
+        # Re-organize scores_dict - for one text, this is just the first score
+        return SequenceClassification._process_predictions(scores_dict, text_idx=0)
+
+    def run_batch(self, texts: List[str]) -> List[ClassificationResult]:
+        """Run the sequence classification on batch, truncates sequences too long for model
+
+        Args:
+            text: List[str]
+                Input strings to be classified
+
+        Returns:
+            List[ClassificationResult]
+        """
+        scores_dict = self._get_scores(texts)
+        num_texts = len(texts)
+
+        # Re-organize scores_dict for each example
+        # We could eventually consider whether or not to sort classifications by scores
+        # but avoiding this prescription here for now
+        classification_predictions = []
+        for text_idx in range(num_texts):
+            classification_prediction = SequenceClassification._process_predictions(
+                scores_dict, text_idx
+            )
+            classification_predictions.append(classification_prediction)
+        return classification_predictions
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        self.tokenizer.save_pretrained(model_path)
+        self.model.save_pretrained(model_path)
+
+    @classmethod
+    def load(cls, model_path: str) -> "SequenceClassification":
+        """Load a tokenizer and sequence classification model. Assumes the
+        tokenizer and model are HuggingFace transformer-based and on
+        the same model_path
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            SequenceClassification
+                Instance of this class built from the on disk model.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(model_path)
+        model = AutoModelForSequenceClassification.from_pretrained(model_path)
+        device = SequenceClassification._get_device()
```

Comment:
A little confused about this feedback since this is the `load` function and we typically only take in `model_path` at load time. I didn't want to make it another runtime arg since I presumed the device would be constant for the instance.

Reply:
Gotcha - I am going to go ahead and remove this for now since the immediate use case does not require this explicit GPU support and will add a comment for future usage to add/test on GPU